gdal-src/build.rs

```diff
@@ -0,0 +1,58 @@
+fn main() {
+    let sqlite3_include_dir =
+        std::env::var("DEP_SQLITE3_INCLUDE").expect("This is set by libsqlite3-sys");
+    let sqlite3_lib_dir = std::env::var("DEP_SQLITE3_LIB_DIR").expect("set by libsqlite3-sys");
+    let proj_root = PathBuf::from(std::env::var("DEP_PROJ_ROOT").expect("set by proj-sys"));
+    let proj_lib = if proj_root.join("lib").join("proj_d.lib").exists() {
+        "proj_d.lib"
+    } else if proj_root.join("lib").join("proj.lib").exists() {
+        "proj.lib"
+    } else {
+        "libproj.a"
+    };
+
+    let res = cmake::Config::new("source")
+        .define("GDAL_BUILD_OPTIONAL_DRIVERS", "OFF")
+        .define("OGR_BUILD_OPTIONAL_DRIVERS", "OFF")
+        .define("GDAL_USE_INTERNAL_LIBS", "ON")
+        .define("GDAL_USE_EXTERNAL_LIBS", "OFF")
+        .define("BUILD_SHARED_LIBS", "OFF")
+        .define("BUILD_STATIC_LIBS", "ON")
+        .define("BUILD_APPS", "OFF")
+        .define("BUILD_DOCS", "OFF")
+        .define("BUILD_TESTING", "OFF")
+        .define("BUILD_GMOCK", "OFF")
+        .define("PROJ_INCLUDE_DIR", format!("{proj_root}/include"))
+        .define("PROJ_LIBRARY", format!("{proj_root}/lib/{proj_lib}"))
+        // enable the gpkg driver
```

Comment:
I require the geopackage driver for my usecase that's why I added these configurations. In the end we already depend on `libsqlite3` via proj, so that should hopefully be fine.

That written: Anyone that tries to actually use that driver to create a geopackage dataset will likely run into: https://github.com/OSGeo/gdal/issues/9135 (which hopefully will be fixed soon upstream, so it might be worth to pull in the fix into the bundled version afterwards.)